### PR TITLE
Update CLAS ZCL_ABAPGIT_USER_MASTER_RECORD

### PR DESCRIPTION
**Pull request by sodogan**
-Issue:If the user does not exist in the system, the BAPI will fail but still the user still will be added to the gt_user. 
-Fixed the issue, if user is invalid it wont be added to the list.
-Refactored the class- added check_user_exist method.
Instead of return code changed to exception for more clean coding.

### CODE REVIEW

**Comment by sodogan** [comment]
I am trying to fix the abaplint issues, i have checked that you use abaplint via VS Code.
Is there a plugin for Eclipse?

File: src/utils/zcl_abapgit_user_master_record.clas.abap [hunk 1]
```diff
@@ -39,76 +33,71 @@ CLASS zcl_abapgit_user_master_record DEFINITION
...
+
+    "Get user details
+    TRY.
+        check_user_exists(
+              EXPORTING
+                iv_user = iv_user
+              IMPORTING
+                es_address   = ls_address
+                et_smtp      = lt_smtp
+                ).
```
Comment (larshp) [inline src/utils/zcl_abapgit_user_master_record.clas.abap:78]:
```suggestion
                et_smtp      = lt_smtp ).
```

File: src/utils/zcl_abapgit_user_master_record.clas.abap [hunk 2]
```diff
@@ -39,76 +33,71 @@ CLASS zcl_abapgit_user_master_record DEFINITION
...
-      ENDLOOP.
+        ENDLOOP.
+        " Attempt to use the full name from SU01
+        ms_user-name = ls_address-fullname.
+      CATCH zcx_abapgit_exception INTO lo_exception.
+        "Could not find user,try to get from other clients
+        get_user_dtls_from_other_clnt( iv_user ).
+    ENDTRY.
+    "if the user has been found successfully ad it to the list
+    IF ( ms_user-name is not INITIAL AND ms_user-email IS NOT INITIAL ).
```
Comment (larshp) [inline src/utils/zcl_abapgit_user_master_record.clas.abap:94]:
```suggestion
    IF ( ms_user-name IS NOT INITIAL AND ms_user-email IS NOT INITIAL ).
```

**Comment by larshp** [comment]
@sodogan see suggested changes above, they will most likely fix the errors reported by the linter

**Comment by larshp** [comment]
@sodogan see suggested changes above

**Comment by sodogan** [comment]
Hi Lars
Thank you for your help i am on holidays 
I will get back to contribute as soon as i come back
Do you prefer to create a feature request before writing the code and doing a pull request?
So if you agree then do the code change 
Please let me know what you prefer

**Comment by sodogan** [comment]
Lars
I have installed abaplint and VScode and checked the validations to see.
Its giving many more errors, i thought its due to the abaplint.json config file(rule configuration file) 
Lars please advise me how to setup abaplint correctly

File: src/zcx_abapgit_exception.clas.abap [hunk 3]
```diff
@@ -97,6 +97,32 @@ CLASS ZCX_ABAPGIT_EXCEPTION IMPLEMENTATION.
   ENDMETHOD.
 
 
+  METHOD get_longtext.
```
Comment (larshp) [inline src/zcx_abapgit_exception.clas.abap:100]: no changes to this class please

**Comment by larshp** [comment]
go to https://github.com/larshp/abapGit/pull/3709/files to see the error message, the error is "Forbidden pragma", remove the pragma to solve the issue

image

**Comment by mbtools** [comment]
Hi @sodogan, thanks for your help with this. We understand that this has become a bit tricky due to abaplint checks and now conflicts. I have taken your suggestions into a new PR https://github.com/abapGit/abapGit/pull/3994. 

Please don't hesitate to contribute in the future!

**Pull request closed without merging**